### ISSUE TITLE
Awareness of `Copy` types.

### ISSUE THREAD

**Issue by LouisGariepy** [LouisGariepy/cornucopia#56]
> Could you clarify please, I'm not sure I'm following 😄. What I've done as of now is to not generate the structs if there are no return columns, for example, but I'm not sure if that's what you're referring to.

I do not want useless borrowed variant:

```sql
--! authors
SELECT name WHERE city = :city FROM Author;

--! authors_age
SELECT age WHERE city = :city FROM Author;
```

```rs
// Borrowed version because &str can be borrowed
pub struct AuthorsBorrowed<'a> {
    name: &'a str,
}

pub struct Authors {
    name: String,
}

impl<'a> From<AuthorsBorrowed<'a>> for Authors {
    fn from(it: AuthorsBorrowed<'a>) -> Self {
        Authors {
            name: it.name.to_owned(),
        }
    }
}

// Only owned because there is nothing to borrow
pub struct AuthorsAge {
    age: i64,
}
```

But thinking about it, handling these cases would increase the complexity of the code, so we should look at how hard it is to implement.

_Originally posted by @Virgiel in https://github.com/LouisGariepy/cornucopia/issues/41#issuecomment-1132276087_

**Comment by LouisGariepy** [comment]
In terms of basic implementation, it'd probably work by having each CornucopiaType type hold a boolean (is_copy or whatever) which would then be used during codegen. Basic types' copy-ability would simply be hard-coded in the registrar, while custom types would be recursively traversed during the registration process.

**Comment by LouisGariepy** [comment]
Thankfully, this was very painless to add and seems to work well enough. Should be good to go for next release :smile: :rocket:

**Comment by LouisGariepy** [comment]
On track to be solved by #52.
